### ISSUE TITLE
agent-esx can only authenticate against local VMware users, not cluster users

### ISSUE THREAD

**Issue by cadegenn** [glpi-project/glpi-agent#259]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

Yes, I know

### Describe the bug

Using `agent-esx` with credential of a VSPHERE cluster user result in the following error :
```
Connection failure: 500 Internal Server Error: Cannot complete login due to an incorrect user name or password.
```
Creating a user at the server level (not part of the cluster), `agent-esx` authentication works as expected.

I tried to connect with following syntaxes to no available : 

- `VSPHERE.LOCAL/user`
- `VSPHERE.LOCAL\user`
- `VSPHERE.LOCAL\\user`
- `user`
- `user@vsphere.local`

none worked.

when finally I created a user at the local level with the local ESXi interface, using `agent-esx --user <myuser>` with no domain prefix works instantly.

### To reproduce

1. install at least 2 VMware ESXi serveurs
2. make them part of the same cluster
3. create a user account on that cluster and give him permission (read only is enough)
4. try to inventory one server with `agent-esx --host <your ESX host> --user vsphere.local/<your user> --password <your password> --directory /tmp` 

### Expected behavior

`agent-esx` is expected to authenticate with cluster user as well as local user.

### Operating system

Linux

### GLPI Agent version

1.4

### GLPI version

10.0.3

### GLPIInventory plugin or FusionInventory for GLPI plugin version

1.0.4

### Additional context

Thank you very much

**Comment by ClementJ35** [comment]
Hi,
I confirm, this may be not a bug but surely is limitating.
Had to create a dedicated user everywhere.

**Comment by g-bougard** [comment]
Hi,
actually, I don't have such environment to test. So I'm really not sure what to do. Anyway, first thing to check, do you have any non standard chars in the user or password ? Maybe this can be a SOAP messages encoding issue.

**Comment by g-bougard** [comment]
Checking few VMware KB entries, I'm curious: are you able to access `/var/log/auth.log` file on the ESX and check what it reports when you try to inventory the ESX with a cluster account ?

**Comment by g-bougard** [comment]
It seems also ESX with block an ip for 120 seconds by default returning a "Cannot complete login due to an incorrect user name or password." message after few authentication failure.
Is it possible you tried the right user syntax after you've been blocked by ESX ? Maybe `/var/log/auth.log` can reveal such context.

**Comment by ClementJ35** [comment]
Despite what I thought, I'm not able to connect to ESXi via the WebUI with those accounts either.
Connecting to vsphere is working with @vsphere.local accounts.
May be it's just VMWare design, I can't find clear documentation on login to ESXi with SSO accounts.

**Comment by ClementJ35** [comment]
BTW, the log indicated : 
```
2022-11-29T16:31:34.187Z error hostd[2847771] [Originator@6876 sub=Default opID=e5c9ec5c] [module:pam_lsass]pam_do_authenticate: error [login:inventaire-glpi@vsphere.local][error code:2]
2022-11-29T16:31:34.187Z error hostd[2847771] [Originator@6876 sub=Default opID=e5c9ec5c] [module:pam_lsass]pam_sm_authenticate: failed [error code:2]
2022-11-29T16:31:34.190Z warning hostd[2847771] [Originator@6876 sub=Default opID=e5c9ec5c] Rejected password for user inventaire-glpi@vsphere.local from 10.7.6.130
2022-11-29T16:31:34.190Z info hostd[2847771] [Originator@6876 sub=Vimsvc.ha-eventmgr opID=e5c9ec5c] Event 41138 : Cannot login inventaire-glpi@vsphere.local@10.7.6.130
```

**Comment by cadegenn** [comment]
Hi,
I have the same log as @ClementJ35 
Curiously, this log is in `/var/log/hostd.log` as `/var/log/authd.log` remains empty.
Tell me if you want me to try anything else
Thank you

**Comment by g-bougard** [comment]
Hi,
actually, we even don't know if logging in with a cluster user is possible.
Do you have any usage where you manage to use a SOAP access with such a user to your ESX ?
Eventually, can I suggest you to contact VMware support so they can tell you if a SOAP access with a cluster user is possible ?

**Comment by g-bougard** [comment]
Closing as nothing happens.
